src/main/kotlin/net/atos/zac/signalering/SignaleringenService.kt

```diff
@@ -0,0 +1,299 @@
+                    if (signalering.type.type === SignaleringType.Type.ZAAK_DOCUMENT_TOEGEVOEGD) {
+                        bronnenBuilder.add(signaleringenZACHelper.getDocument(signalering.detail))
+                    }
+                }
+
+                SignaleringSubject.TAAK -> bronnenBuilder.add(signaleringenZACHelper.getTaak(signalering.subject))
+                SignaleringSubject.DOCUMENT -> bronnenBuilder.add(
+                    signaleringenZACHelper.getDocument(signalering.subject)
+                )
+            }
+            mailService.sendMail(
+                MailGegevens(from, to, mailTemplate.onderwerp, mailTemplate.body),
+                bronnenBuilder.build()
+            )
+        }
+    }
+
+    fun createUpdateOrDeleteInstellingen(instellingen: SignaleringInstellingen): SignaleringInstellingen? {
+        ValidationUtil.valideerObject(instellingen)
+        if (instellingen.isEmpty) {
+            if (instellingen.id != null) {
+                entityManager.remove(entityManager.find(SignaleringInstellingen::class.java, instellingen.id))
+            }
+            return null
+        }
+        return entityManager.merge(instellingen)
+    }
+
+    fun readInstellingenGroup(type: SignaleringType.Type?, target: String?): SignaleringInstellingen {
+        val signalering = signaleringInstance(type)
+        signalering.setTargetGroup(target)
+        return readInstellingen(signalering)
+    }
+
+    fun readInstellingenUser(type: SignaleringType.Type?, target: String?): SignaleringInstellingen {
+        val signalering = signaleringInstance(type)
+        signalering.setTargetUser(target)
+        return readInstellingen(signalering)
+    }
+
+    fun readInstellingen(signalering: Signalering): SignaleringInstellingen {
+        val instellingen = listInstellingen(
+            SignaleringInstellingenZoekParameters(signalering)
+        )
+        if (instellingen.size == 1) {
+            return instellingen[0]
+        }
+        return SignaleringInstellingen(signalering.type, signalering.targettype, signalering.target)
+    }
+
+    fun listInstellingen(parameters: SignaleringInstellingenZoekParameters): List<SignaleringInstellingen> {
+        val builder = entityManager.criteriaBuilder
+        val query = builder.createQuery(
+            SignaleringInstellingen::class.java
+        )
+        val root = query.from(
+            SignaleringInstellingen::class.java
+        )
+        return entityManager.createQuery(
+            query.select(root)
+                .where(signaleringenPredicateHelper.getSignaleringInstellingenWhere(parameters, builder, root))
+        )
+            .resultList
+    }
+
+    fun listInstellingenInclusiefMogelijke(
+        parameters: SignaleringInstellingenZoekParameters
+    ): List<SignaleringInstellingen> {
+        val map = listInstellingen(parameters).stream()
+            .collect(
+                Collectors.toMap(
```

Comment:
I'm pretty sure `Collectors.toMap` is not needed but maybe improve this in a future PR.. :-)